### ISSUE TITLE
Notification: "A new PowerShell stable release is available: …" - a small but most useful improvement

### ISSUE THREAD

**Issue by schittli** [PowerShell/PowerShell#16522]
### Summary of the new feature / enhancement

Good morning

it's great to have many PowerShell users - it's a great 'tool', thanks a lot for your great work!

### … but it's **very** annoying, that all those PowerShell users get this notification:
```
   A new PowerShell stable release is available: …
   Upgrade now, or check out the release page at:
   https://aka.ms/PowerShell-Release?tag=…
```

### The problem is, that most Users will start this process:
1. _"A new release? - great!"_
2. But: _"How can I upgrade PowerShell??"_
3. _"… ok, I'll upgrade it anytime later, I don't have time to ask google"_

### Therefore: please tell how PowerShell can be upgraded in the upgrade notification!

Something like this:
```
   A new PowerShell stable release is available: …
   Start upgrade by calling: …
   The Release Notes: https://aka.ms/PowerShell-Release?tag=…
```

Thanks a lot, kind regards,
Thomas

**Comment by iSazonov** [comment]
It is not clear that you ask. The link in the notice open a page where you can download PowerShell distributive from.

**Comment by rkeithhill** [comment]
As @iSazonov points out, that link takes you to the downloads page for the updated version of PowerShell.  However, there are a number of ways PowerShell could be installed.  So the update process isn't really a single update process as it depends on how you installed PowerShell: direct MSI download, Windows Store, winget, Chocolatey, dotnet global tool, apt, yum, snap, homebrew, etc.  

Related to #8663

**Comment by doctordns** [comment]
We can do better.

**Comment by m4heshd** [comment]
Can't remember how I installed it in the first place (since there are 100 million ways to install powershell) and have no idea how to update it now. 😔

**Comment by doctordns** [comment]
Google is your friend! 

You have options. You can go to github to download the install-poweeshelll.ps1 script and  use that. Or, assuming youn enabled update by MS update just wait. It seems to take MSFT to get the updates from GitHub go MU (do they use the USPS??) So be patient

**Comment by m4heshd** [comment]
> Or, assuming you enabled update by MS update just wait

I actually realized this right after I left the comment because I downloaded the installer from the latest release and my previous download was there. Then when I ran the installer once more, I've realized I enabled updates via WU the last time I've installed it. PS seems to have updated once or twice since then. 😁 So all good. Thanks for the response.

**Comment by leonheess** [comment]
How can I upgrade??

**Comment by Science001** [comment]
If using `winget`, you can upgrade easily by just executing `winget upgrade -q Powershell` from powershell itself.

**Comment by RobMahn** [comment]
My message says:
 "A new PowerShell stable release is available: v7.3.1
   Upgrade now, or check out the release page at:
     https://aka.ms/PowerShell-Release?tag=v7.3.1"

This indicates that the link is to "check out the release", not a process to upgrade.  This is also a logical 'or' to upgrade now, so I was not incented to look there for the upgrade.

**Comment by Will-at-FreedomDev** [comment]
> If using `winget`, you can upgrade easily by just executing `winget upgrade -q Powershell` from powershell itself.

I was hoping that this was the way to do it, but I got the output "No applicable upgrade found.". Even though when I start PowerShell it literally says to "Upgrade now" lol

**Comment by Will-at-FreedomDev** [comment]
> install-powershell.ps1
This worked great. It took me a few minutes to find this issue then go find the file (the original commenter misspelled the filename :) )

It makes A LOT of sense for the message to say "run this command to download and execute the install script" instead of needing to hunt for it as others have mentioned as well.

**Comment by fatwreck** [comment]
> If using `winget`, you can upgrade easily by just executing `winget upgrade -q Powershell` from powershell itself.

I get the same result as @Will-at-FreedomDev running this.

- `install-powershell.ps1` isn't recognized as a name of a cmdlet etc.
- I've downloaded and installed from the MS store, but after restarting pwsh and computer I still have the upgrade notice.
- I've followed and ran the instructions on this blog post (https://www.thomasmaurer.ch/2019/07/how-to-install-and-update-powershell-7/)
- I've downloaded the update from here: https://github.com/PowerShell/PowerShell/releases/tag/v7.3.2 but the folder has no clear direction for what to do with it. There's 357 files in the root and no readme. Run pwsh.exe? Runs, but still not updating my local install.
- Github page has no instructions

I'm just trying to update to get past a bug preventing me from running a script and getting on with the day. Next step is looking for the `Install-PowerShell.ps1` mentioned above and crossing my fingers it'll do something.

Might not be an issue for people who have experience upgrading, but for the rest of us there's a lot of room to improve this obtuse process.

**Comment by Will-at-FreedomDev** [comment]
@fatwreck sorry, I should have posted it for others looking for it. At the moment, it can be found here: https://github.com/PowerShell/PowerShell/blob/master/tools/install-powershell.ps1

**Comment by langemol** [comment]
The correct winget command would be 
`winget upgrade --id Microsoft.PowerShell`
But that doesn't really help because it returns
>A newer version was found, but the install technology is different from the current version installed. Please uninstall the package and install the newer version.

**Comment by GitMurf** [comment]
I still cannot believe this has not been addressed :( Had to spend 20 minutes googling AFTER I ignored updating for several days because I didn't want to have to deal with figuring out how to upgrade. I highly suggest you at minimum change the verbiage of the message. When you say "upgrade now" OR go checkout the release notes, one would assume there would be a clear and easy way to "click something" or "run some command" to upgrade now and that it would be provided there in the notice.

Given there are different ways to install PowerShell, if you are not able to provide a single command (like winget) that will show the user how to upgrade immediately, then at minimum you need to create a "how to upgrade" page that you can link the URL just like you link the Release Notes in the message. At that "how to upgrade" page it can show you the several different ways to upgrade depending on how you installed.

For me for example, I didn't even remember how I installed it originally, so I just tried the `winget` method from this comment from @Science001 (Thanks!) (https://github.com/PowerShell/PowerShell/issues/16522#issuecomment-1350663047) and it happened to work. You cannot expect most people know / remember how they initially installed PowerShell and to have so many different ways to upgrade that you cannot provide a single simple command, seems like a problem to me.

**Comment by molodchyk** [comment]
I got to that link and downloaded PowerShell-7.3.3-win-x64.msi, worked really well. But yes I agree, the message could be a bit  clearer maybe.

**Comment by doctordns** [comment]
The file Install-PowerShell.ps1 is found at https://github.com/PowerShell/PowerShell/blob/master/tools/install-powershell.ps1

It works to install PowerShell 7, update to a later version, or install either preview or daily build. I have been using this for several years, and it always works.

**Comment by sruPL** [comment]
This could be much more understandable and user-friendly. As above, just include this note:

Run "winget upgrade -q Powershell" to update.

**Comment by stanoychev** [comment]
quality work 👌

**Comment by kei-kusanagi** [comment]
I have same problem.

I go to Microsoft store and below the open button there was a button that says update, click in update ands this works from me (sorry mi bad spanglish :p )
![image](https://github.com/PowerShell/PowerShell/assets/93227096/1eb31729-e467-4a7d-ac13-30b5266fc944)

**Comment by systembugbd** [comment]
goto terminal and paste below code and hit enter.
Invoke-Expression "& { $(Invoke-RestMethod https://aka.ms/install-powershell.ps1) } -UseMSI"

will install the power shell latest version and solve the issues.

**Comment by gabrielecastellano** [comment]
The issue was closed silently but the message is still the same.

**Comment by Chiramisu** [comment]
It would sure be great if the `upgrade` command worked properly, but in my experience it hasn't for this particular package, so I typically run `winget install --force Microsoft.PowerShell` instead (as Admin because I `--scope machine`).

To the OP's point, since I use Winget, I sometimes get a notification of a new version in my Terminal before it's actually available in the Winget package catalog, so it would be nice to defer that notification for a user-specified period. For example, two weeks after release, THEN notify; Alternatively, only notify me when the update is actually available through Winget.

**Comment by scarolan** [comment]
This is super annoying.  Please include an option for powershell to upgrade itself.

**Comment by saucyknave** [comment]
I realize it's utterly futile to comment here, but I'm compelled anyway.  I just ran into this message this morning, on PowerShell 7.4.5.  After reading all the comments, I ran the winget command (winget upgrade -q PowerShell), which SUCCESSFULLY upgraded my PowerShell.  What's amusing is that I remained in the PowerShell console currently open where I ran the winget command, and at the end of the upgrade, it output "UPGRADE ABANDONED" onto the command line.  I closed and reopened PowerShell and it shows "PowerShell 7.5.0", so I know it was a successful upgrade, but I think it's interesting that if you stay in the console while upgrading, it'll tell you it "abandoned" the upgrade.
